File: source/kCura.Relativity.Export/Exporters/ILongTextStreamFormatter.vb

Imports kCura.Relativity.Export.FileObjects

Namespace kCura.Relativity.Export.Exports
	Public Interface ILongTextStreamFormatter
		Sub TransformAndWriteCharacter(ByVal character As Int32, ByVal outputStream As System.IO.TextWriter)
	End Interface
	Public Class NonTransformFormatter
		Implements ILongTextStreamFormatter
		Public Sub TransformAndWriteCharacter(ByVal character As Integer, ByVal outputStream As System.IO.TextWriter) Implements ILongTextStreamFormatter.TransformAndWriteCharacter
			outputStream.Write(ChrW(character))
		End Sub
	End Class
	Public Class HtmlFileLongTextStreamFormatter
		Implements ILongTextStreamFormatter

		Private _source As System.IO.TextReader

		Public Sub New(ByVal settings As ExportFile, ByVal source As System.IO.TextReader)
			_source = source
		End Sub

		Public Sub TransformAndWriteCharacter(ByVal character As Integer, ByVal outputStream As System.IO.TextWriter) Implements ILongTextStreamFormatter.TransformAndWriteCharacter
			Select Case character
				Case 13
					outputStream.Write("<br/>")
					If _source.Peek = 10 Then _source.Read()
				Case 10
					outputStream.Write("<br/>")
				Case Else
					outputStream.Write(System.Web.HttpUtility.HtmlEncode(ChrW(character)))
			End Select
		End Sub

	End Class

	Public Class DelimitedFileLongTextStreamFormatter
		Implements ILongTextStreamFormatter
		Private _quoteDelimiter As Char
		Private _newlineDelimiter As Char
		Private _source As System.IO.TextReader

		Public Sub New(ByVal settings As ExportFile, ByVal source As System.IO.TextReader)
			_quoteDelimiter = settings.QuoteDelimiter
			_newlineDelimiter = settings.NewlineDelimiter
			_source = source
		End Sub

		Public Sub TransformAndWriteCharacter(ByVal character As Int32, ByVal outputStream As System.IO.TextWriter) Implements ILongTextStreamFormatter.TransformAndWriteCharacter
			Select Case character
				Case AscW(_quoteDelimiter)
					outputStream.Write(_quoteDelimiter & _quoteDelimiter)
				Case 13, 10
					outputStream.Write(_newlineDelimiter)
					If _source.Peek = 10 Then
						_source.Read()
					End If
				Case Else
					outputStream.Write(ChrW(character))
			End Select
		End Sub

	End Class


End Namespace

File: source/kCura.Relativity.Export.ConvertedToC#/Exporters/ILongTextStreamFormatter.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Data;
using System.Linq;

using kCura.Relativity.Export.FileObjects;

namespace kCura.Relativity.Export.Exports
{
    public interface ILongTextStreamFormatter
    {
        void TransformAndWriteCharacter(Int32 character, System.IO.TextWriter outputStream);
    }
    public class NonTransformFormatter : ILongTextStreamFormatter
    {
        public void TransformAndWriteCharacter(int character, System.IO.TextWriter outputStream)
        {
            outputStream.Write(Strings.ChrW(character));
        }
    }
    public class HtmlFileLongTextStreamFormatter : ILongTextStreamFormatter
    {


        private System.IO.TextReader _source;
        public HtmlFileLongTextStreamFormatter(ExportFile settings, System.IO.TextReader source)
        {
            _source = source;
        }

        public void TransformAndWriteCharacter(int character, System.IO.TextWriter outputStream)
        {
            switch (character)
            {
                case 13:
                    outputStream.Write("<br/>");
                    if (_source.Peek() == 10)
                        _source.Read();
                    break;
                case 10:
                    outputStream.Write("<br/>");
                    break;
                default:
                    outputStream.Write(System.Web.HttpUtility.HtmlEncode(Strings.ChrW(character)));
                    break;
            }
        }

    }

    public class DelimitedFileLongTextStreamFormatter : ILongTextStreamFormatter
    {
        private char _quoteDelimiter;
        private char _newlineDelimiter;

        private System.IO.TextReader _source;
        public DelimitedFileLongTextStreamFormatter(ExportFile settings, System.IO.TextReader source)
        {
            _quoteDelimiter = settings.QuoteDelimiter;
            _newlineDelimiter = settings.NewlineDelimiter;
            _source = source;
        }

        public void TransformAndWriteCharacter(Int32 character, System.IO.TextWriter outputStream)
        {

            if (character == Strings.AscW(_quoteDelimiter))
            {
                outputStream.Write(_quoteDelimiter + _quoteDelimiter);
            }
            else if (character == 13 || character == 10)
            {
                outputStream.Write(_newlineDelimiter);
                if (_source.Peek() == 10)
                {
                    _source.Read();
                }
            }
            else {
                outputStream.Write(Strings.ChrW(character));
            }
        }

    }


}

